packages/libs/hd-wallet/src/utils/encrypt.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  createCipheriv,
+  createDecipheriv,
+  pbkdf2Sync,
+  randomBytes,
+} from 'crypto';
+
+type BinaryLike = string | NodeJS.ArrayBufferView;
+
+const SALT = Buffer.from('FkM5B23nB6LVY7mrwDeh'); // SALT value used for deriving keys
+/**
+ * Derive a cryptographic key from the provided password.
+ * @param {string} password - User's password.
+ * @returns {Buffer} - Returns the derived cryptographic key.
+ */
+function deriveKey(password: string, salt: BinaryLike): Buffer {
+  return pbkdf2Sync(password, salt, 1000, 32, 'sha256');
+}
+
+/**
+ * Encrypt the provided text using AES-256-GCM algorithm.
+ * @param {Buffer} text - Text to encrypt.
+ * @param {string} password - User's password.
+ * @returns {{ cipherText: Buffer; iv: Buffer; tag: Buffer }} - Returns the encrypted text, initialization vector, and authentication tag.
+ */
+export function encrypt(
+  text: Buffer,
+  password: string,
+  salt: BinaryLike,
+): { cipherText: Buffer; iv: Buffer; tag: Buffer } {
+  const key = deriveKey(password, salt);
+  const iv = randomBytes(12);
+  const cipher = createCipheriv('aes-256-gcm', key, iv);
+  const cipherText = Buffer.concat([cipher.update(text), cipher.final()]);
+  const tag = cipher.getAuthTag(); // Capture the authentication tag
+  return {
+    cipherText,
+    iv,
+    tag,
+  };
+}
+
+type Encrypted = ReturnType<typeof encrypt>;
+
+export function encryptPrivateKey(
+  privateKey: Uint8Array,
+  password: string,
+): string {
```

Comment:
I'm not sure whether this should be part of this lib. The lib's responsibility seems to me to be only hd-wallet, just mnemonics and it's derivation and security? Let's discuss.